src/gui/accountsettings.cpp

```diff
@@ -1368,6 +1368,61 @@ void AccountSettings::slotSelectiveSyncChanged(const QModelIndex &topLeft,
     }
 }
 
+void AccountSettings::slotPossiblyUnblacklistE2EeFoldersAndRestartSync()
+{
+    disconnect(_unblackListE2eeFoldersOnE2eeInitializedConnection);
+    _unblackListE2eeFoldersOnE2eeInitializedConnection = {};
+
+    for (const auto folder : FolderMan::instance()->map()) {
+        if (folder->accountState() != _accountState) {
+            continue;
+        }
+        bool ok = false;
+        const auto foldersToRemoveFromBlacklist = folder->journalDb()->getSelectiveSyncList(SyncJournalDb::SelectiveSyncE2eFoldersToRemoveFromBlacklist, &ok);
+        if (foldersToRemoveFromBlacklist.isEmpty()) {
+            continue;
+        }
+        auto blackList = folder->journalDb()->getSelectiveSyncList(SyncJournalDb::SelectiveSyncBlackList, &ok);
+        const auto blackListSize = blackList.size();
+        if (blackListSize == 0) {
+            continue;
+        }
+        for (const auto pathToRemoveFromBlackList : foldersToRemoveFromBlacklist) {
+            blackList.removeAll(pathToRemoveFromBlackList);
+        }
+        if (blackList.size() != blackListSize) {
+            const auto updateBlackListAndScheduleFolderSync = [blackList, folder, foldersToRemoveFromBlacklist]() {
+                folder->journalDb()->setSelectiveSyncList(SyncJournalDb::SelectiveSyncBlackList, blackList);
+                folder->journalDb()->setSelectiveSyncList(SyncJournalDb::SelectiveSyncE2eFoldersToRemoveFromBlacklist, {});
+                for (const auto pathToRemoteDiscover : foldersToRemoveFromBlacklist) {
+                    folder->journalDb()->schedulePathForRemoteDiscovery(pathToRemoteDiscover);
+                }
+                FolderMan::instance()->scheduleFolder(folder);
+            };
```

Comment:
Since this doesn't need to be passed as a callback to anything, I would move it out into a private slot